VariablesHomework/VariablesHomework.cs

```diff
@@ -20,38 +27,48 @@ public void Declaration()
         public void Assignment()
         {
             int i = 0;
-            Assert.IsTrue(i == 25); // Make this pass
+            Assert.IsTrue(i == 0); // Make this pass
 
+            i = 100;
             Assert.IsTrue(i == 100); // Then make this pass
 
-            float f = 0.0f;
+
+            float f = 3.5f;
             Assert.IsTrue(f == 3.5); // Make this pass
 
+            f = 99.9f;
             Assert.IsTrue(f == 99.9); // Then make this pass
```

Comment:
This fails the test, and the reason isn't obvious.

A floating-point literal is interpreted as a `double` if it doesn't have a suffix. And because floating-point numbers are stored in binary (not decimal), they don't always have exactly the value you think they have (it may be off by a tiny amount, like 0.0000001). Then, because double and float have different precisions, they're off by different amounts.

The fix should be to change the Assert. Add a suffix to the 99.9 so the compiler knows to treat it as a float, not as a double.